src/components/opt-in-out/helpers/post-form-data.ts

```diff
@@ -0,0 +1,27 @@
+import { allowedKeys } from 'pages/api/opt-out-feedback'
+
+type AllowedFormKeys = typeof allowedKeys[number]
+
+type OptOutData = {
+  [key in AllowedFormKeys]: string | null
+}
+
+export async function postFormData(data: OptOutData) {
+  try {
+    const res = await fetch('/api/opt-out-feedback', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    })
+
+    if (res.status === 204 || res.status === 200) {
+      process.env.NODE_ENV !== 'production' && console.log('Feedback submitted')
+    } else {
+      console.warn(`${res.status}: Unexpected result from feedback api`)
+    }
+  } catch (e) {
```

Comment:
I figure we should catch everything as we don't want a borked form to bring down the page? Lmk if I should handle this differently.

Reply:
Oh cool you are catching here 👍  disregard my other comment!